tests/mdev_configuration_allocation_test.go

```diff
@@ -354,6 +360,174 @@ var _ = Describe("[Serial][sig-compute]MediatedDevices", Serial, decorators.VGPU
+				)
+				Expect(err).ToNot(HaveOccurred())
+
+				resvHugepages, err := strconv.Atoi(strings.Trim(output, "\n"))
+				Expect(err).ToNot(HaveOccurred())
+
+				// Verify that the VM memory equals to a number of consumed hugepages
+				vmHugepagesConsumption := int64(totalHugepages-freeHugepages+resvHugepages) * hugepagesSize.Value()
+				vmMemory := hugepagesVmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory]
+				if hugepagesVmi.Spec.Domain.Memory != nil && hugepagesVmi.Spec.Domain.Memory.Guest != nil {
+					vmMemory = *hugepagesVmi.Spec.Domain.Memory.Guest
+				}
+
+				if vmHugepagesConsumption == vmMemory.Value() {
+					return true
+				}
+				return false
+			}
+
+			By("Creating a Fedora VMI")
+			vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse("1G")
+			vGPUs := []v1.GPU{
+				{
+					Name:       "gpu1",
+					DeviceName: deviceName,
+				},
+			}
+			vmi.Spec.Domain.Devices.GPUs = vGPUs
+
+			hugepageSize := "2Mi"
+			memory := "64Mi"
+			hugepageType := k8sv1.ResourceName(k8sv1.ResourceHugePagesPrefix + hugepageSize)
+			v, err := cluster.GetKubernetesVersion()
+			Expect(err).ShouldNot(HaveOccurred())
+			if strings.Contains(v, "1.16") {
+				hugepagesVmi.Annotations = map[string]string{
+					v1.MemfdMemoryBackend: "false",
+				}
+				log.DefaultLogger().Object(hugepagesVmi).Infof("Fall back to use hugepages source file. Libvirt in the 1.16 provider version doesn't support memfd as memory backend")
+			}
+
+			nodeWithHugepages := libnode.GetNodeWithHugepages(virtClient, hugepageType)
+			if nodeWithHugepages == nil {
+				Skip(fmt.Sprintf("No node with hugepages %s capacity", hugepageType))
+			}
+
+			hugepagesVmi.Spec.Affinity = &k8sv1.Affinity{
+				NodeAffinity: &k8sv1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &k8sv1.NodeSelector{
+						NodeSelectorTerms: []k8sv1.NodeSelectorTerm{
+							{
+								MatchExpressions: []k8sv1.NodeSelectorRequirement{
+									{Key: "kubernetes.io/hostname", Operator: k8sv1.NodeSelectorOpIn, Values: []string{nodeWithHugepages.Name}},
+								},
+							},
+						},
+					},
+				},
+			}
+			hugepagesVmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse(memory)
+
+			hugepagesVmi.Spec.Domain.Memory = &v1.Memory{
+				Hugepages: &v1.Hugepages{PageSize: hugepageSize},
+			}
+
+			hugepagesVmi.Spec.Domain.Resources = v1.ResourceRequirements{}
+
+			createdVmi, err := virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(hugepagesVmi)).Create(context.Background(), vmi)
+			Expect(err).ToNot(HaveOccurred())
+			hugepagesVmi = createdVmi
+			libwait.WaitForSuccessfulVMIStart(hugepagesVmi)
+			Expect(console.LoginToFedora(hugepagesVmi)).To(Succeed())
+
+			By("Checking that the VM memory equals to a number of consumed hugepages")
+			Eventually(func() bool { return verifyHugepagesConsumption() }, 30*time.Second, 5*time.Second).Should(BeTrue())
```

Comment:
This is weird... Looks like you're expecting the function itself (not its return value) to become true. Does that work?
And if it does, why not just:
```suggestion
			Eventually(verifyHugepagesConsumption, 30*time.Second, 5*time.Second).Should(BeTrue())
```